vcd/resource_vcd_nsxv_dhcp_relay.go

```diff
@@ -0,0 +1,434 @@
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+var relayAgentResource = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"org_network": {
+			Required:    true,
+			Type:        schema.TypeString,
+			Description: "Org network which is to be used for relaying DHCP message to specified servers",
+		},
+		"gateway_ip_address": {
+			Optional:    true,
+			Computed:    true,
+			Type:        schema.TypeString,
+			Description: "Optional gateway IP address of org network which is to be used for relaying DHCP message to specified servers",
+		},
+	},
+}
+
+func resourceVcdNsxvDhcpRelay() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdNsxvDhcpRelayCreate,
+		Read:   resourceVcdNsxvDhcpRelayRead,
+		Update: resourceVcdNsxvDhcpRelayUpdate,
+		Delete: resourceVcdNsxvDhcpRelayDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdNsxvDhcpRelayImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name for DHCP relay settings",
+			},
+			"ip_addresses": {
+				Optional:    true,
+				Type:        schema.TypeSet,
+				Description: "A set of IP address of DHCP servers",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"domain_names": {
+				Optional:    true,
+				Type:        schema.TypeSet,
+				Description: "A set of IP domain names of DHCP servers",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"ip_sets": {
+				Optional:    true,
+				Type:        schema.TypeSet,
+				Description: "A set of IP set names which consist DHCP servers",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"relay_agent": {
+				Required: true,
+				MinItems: 1,
+				Type:     schema.TypeSet,
+				Elem:     relayAgentResource,
+			},
+		},
+	}
+}
+
+// resourceVcdNsxvDhcpRelayCreate sets up DHCP relay configuration as per supplied schema
+// configuration
+func resourceVcdNsxvDhcpRelayCreate(d *schema.ResourceData, meta interface{}) error {
+	vcdClient := meta.(*VCDClient)
+	vcdClient.lockParentEdgeGtw(d)
+	defer vcdClient.unLockParentEdgeGtw(d)
+
+	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	_, vdc, err := vcdClient.GetOrgAndVdcFromResource(d)
+	if err != nil {
+		return fmt.Errorf(errorRetrievingOrgAndVdc, err)
+	}
```

Comment:
this should be part of `getDhcpRelayType`

Reply:
But I guess this saves nothing. I would have to pass `vcdClient` instead of `vdc` then.

Reply:
Let it be. Changed.